src/wzapi.cpp

```diff
@@ -288,6 +288,7 @@ std::string wzapi::translate(WZAPI_PARAMS(std::string str))
 //--
 int32_t wzapi::syncRandom(WZAPI_PARAMS(uint32_t limit))
 {
+	if (limit == 0) return 0;
```

Comment:
And if limit<0 check in another place?

Reply:
The input is `uint32_t` (unsigned 32-bit integer), so by definition it can't be < 0.

If you try to pass in a negative integer from JS, it probably gets turned into a very large positive number. So I'd say "don't do that" would be the fix. 😄

(The key thing here was fixing the crash.)

Reply:
Calling functions with incorrect arguments is of course a modder's mistake. But it would be nice if the error would be visible in the logs, and not magical incomprehensible behavior in the script.

This applies to all APIs. An inexperienced developer often does not understand what the error is.

Reply:
This is definitely something we can revisit in the future, but it may benefit from some modifications to QuickJS itself, so it will likely be further out (and not the immediate next release).